lib/validator.js

```diff
@@ -0,0 +1,47 @@
+const {
+  getCaster
+} = require('./types.js');
+
+class Validator {
+  constructor(keyValue, { type, required }) {
+    this.keyValue = keyValue,
+    this.type = type,
+    this.required =  required;
+  }  
+
+  validate(object) {
+      
+    if(this.required && !(object[this.keyValue])) {
+      throw new Error('error');
+    }
+
+    if(!this.required && !(object[this.keyValue])) {
+      return 'Not Required';
+    }
+
+    const objectType = getCaster(this.type);
+    return objectType(object[this.keyValue]);
```

Comment:
looks good. I would call this variable caster or something. Calling it objectType is confusing